server.go

```diff
@@ -93,6 +77,85 @@ func addItem(c *gin.Context) {
 }
 
 func getItems(c *gin.Context) {
+	sqlInfo := fmt.Sprintf("host=%s port=%d user=%s "+
+	"password=%s dbname=%s sslmode=disable",
+	host, port, user, password, dbname)
+	db, err := sql.Open("postgres", sqlInfo)
+	if err != nil {
+		response := Response{Action: "SQL", Sucessful: false, Context: err.Error()}
+		c.IndentedJSON(http.StatusOK, response)
+	}
+	db.Ping()
+
+	rows, sizeerr := db.Query("SELECT * FROM items")
+	if (sizeerr != nil) {
+		response := Response{Action: "SQL", Sucessful: false, Context: sizeerr.Error()}
+		c.IndentedJSON(http.StatusOK, response)	
+	}
+	defer rows.Close()
+
+	nameFilter, nameFilterPresent := c.GetQuery("rawName")
+	numFilter := c.Param("id")
+	intNumFilter, _ := strconv.Atoi(c.Param("id"))
+	if intNumFilter < 0 {
+		response := Response{Action: "Get", Sucessful: false, Context: "ID cannot be below 0"}
+		c.IndentedJSON(http.StatusOK, response)
+	}
+	var numFilterPresent bool = (numFilter != "")
+	var collection Collection
+	size := 0
+	for rows.Next() {
+		validEntry := true
+		var entry Item
+		rows.Scan(&entry.ID, &entry.Name, &entry.Desc, &entry.TopPriority, &entry.Completed)
+		if (nameFilterPresent) {
+			if (entry.Name != nameFilter) {
+				validEntry = false
+			}
+		}
+		if (numFilterPresent) {
+			if (entry.ID != intNumFilter) {
+				validEntry = false
+			}
+		}
+		if (validEntry) {
+			size++
+		}
+	}
+	newRows, sizeerr := db.Query("SELECT * FROM items")
+	if (sizeerr != nil) {
+		response := Response{Action: "SQL", Sucessful: false, Context: sizeerr.Error()}
+		c.IndentedJSON(http.StatusOK, response)	
+	}
+	defer newRows.Close()
+
+	index := 0
+	collection.Items = make([]Item, size)
+	for newRows.Next() {
+		var entry Item
+		var validEntry bool = true
+		newRows.Scan(&entry.ID, &entry.Name, &entry.Desc, &entry.TopPriority, &entry.Completed)
+		if (nameFilterPresent) {
+			if (entry.Name != nameFilter) {
+				validEntry = false
+			}
+		}
+		if (numFilterPresent) {
+			columnID, _ := strconv.Atoi(numFilter)
+			if (entry.ID != columnID) {
+				validEntry = false
+			}
+		}
+		if (validEntry) {
+			collection.Items[index] = entry
+			index++
+		}
+	}
+
+	c.IndentedJSON(http.StatusOK, collection)
+}
+
+/*func getItems(c *gin.Context) {
```

Comment:
Lets add some comments throughout this section to explain what some of these chunks of code are doing! Large blocks of code without explanation can be tough to come back to

Reply:
//Ok comments it is

Reply:
Great comments. Somewhere between this level and nothing is usually ideal